packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApi.js

```diff
@@ -0,0 +1,306 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+// Note: should really extend DhisApi but there's some sort of circular dependency issue
+
+// TODO: move to actual db table
+const TABLE_DHIS_ORG_UNIT_MAPPING = {
+  TO: 'jSk4k1gnSqk',
+};
+
+// TODO: move to actual db table
+const DATA_SOURCE_CONFIG_BY_CODE = {
+  // SS162: { dhisId: 'pDhCn9lbi5v' },
+  // SS128: { dhisId: 'K3wJsRJHMCa' },
+  // SS182: { dhisId: 'g2KqfrYXLdM' },
+  // SS190A: { dhisId: 'iYzYjNzDnAY' },
+  // SS192: { dhisId: 'RngB7fLGI3i' },
+  // SS219: { dhisId: 'bcg0ApKZTav' },
+  // DP9: { dhisId: 'OzDb71SXJgP' },
+  HP02: { dhisId: 'OmssK27WEJB' },
+};
+
+export class DhisInputSchemeResolvingApi {
+  constructor(models, api) {
+    this.models = models;
+    this.api = api;
+  }
+
+  async getAnalytics(query) {
+    let modifiedQuery = { ...query };
+
+    // console.log('DhisInputSchemeResolvingApi', 'getAnalytics', 'query', query);
+
+    if (
```

Comment:
I checked DHIS api docs, couldn't find a way of specifying the inputIdScheme individually per dimension, and tried a few different ways in postman, I think it has to be all or nothing here...

Reply:
Could you check whether the schemes described in https://docs.dhis2.org/2.28/en/developer/html/dhis2_developer_manual_full.html#webapi_identifier_schemes apply to the endpoints we use?

Reply:
Ah there that is, thanks. No luck unfortunately...

I had a look at the request class for this in dhis2 and it only seems to have inputIdSchema:

https://github.com/dhis2/dhis2-core/blob/d883e68107f971f00f7b9f42d6dd7b001d6cbfd1/dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/AggregateAnalyticsQueryCriteria.java#L182